pymc3/distributions/continuous.py

```diff
@@ -141,22 +286,85 @@ class Normal(Continuous):
     """
     def __init__(self, mu=0.0, tau=None, sd=None, *args, **kwargs):
         super(Normal, self).__init__(*args, **kwargs)
-        self.mean = self.median = self.mode = self.mu = mu
-        self.tau, self.sd = get_tau_sd(tau=tau, sd=sd)
+        self.mu, self.tau = self.parametrize(mu, tau, sd)
+        self.mean = self.median = self.mode = self.mu
         self.variance = 1. / self.tau
 
-    def logp(self, value):
+    def parametrize(self, mu=None, tau=None, sd=None):
+        try:
+            tau, sd = get_tau_sd(tau, sd)
+        except ValueError as err:
+            if check:
+                raise err
+        return mu, tau
+    
+    def pdf(self, x, point=None):
+        """
+        The probability density function (pdf) of the Normal distribution.
+        
+        Parameters
+        ----------
+            x : float
+                The value at which to evaluate the pdf.
+            point : dict
+                Parameter values to fix.                
+        """
+        mu, tau = draw_values([self.mu, self.tau], point=point)                                     
+        return np.sqrt(0.5 * tau / pi) * np.exp(-0.5 * tau * (x - mu) ** 2) 
+    
+    
+    def cdf(self, x, point=None):
+        """
+        The cumulative distribution function (cdf) of the Normal distribution.
+        
+        Parameters
+        ----------
+            x : float
+                The value at which to evaluate the cdf.
+            point : dict
+                Parameter values to fix.                
+        """
+        mu, tau = draw_values([self.mu, self.tau], point=point)   
+        return 0.5 * (1. + sp.erf(np.sqrt(0.5 * tau) * (x - mu)))
+    
+    
+    def random(self, size=None, point=None):
+        """
+        Generate samples from the Normal distribution.
+        
+        Parameters
+        ----------
+        size : integer
+            The number (or array shape) of samples to generate. If not specified, returns one sample.
+        point : dict
+            Model parameters which are to be fixed.
+        """
+        mu, tau = draw_values([self.mu, self.tau], point=point)   
+        return nr.normal(loc=mu, scale=1./tau, size=size)
+    
+    def logp(self, x):
+        """
+        Return the log-likelihood of x given the current
+        parameters of the distribution. 
+        
+        This function is optimised by theano.        
+        
+        Parameters
+        ----------
+        x : float
+            The iid x(s) to evaluate the log-likelihood. 
+        """
         tau = self.tau
-        sd = self.sd
```

Comment:
Actually enforcing `sd > 0` is required because `sd` could be negative and yet `tau` would be positive.